File: backend/src/main/java/com/server/backend/question/QuestionMapper.java
package com.server.backend.question;

import java.util.ArrayList;
import java.util.HashSet;

public final class QuestionMapper {

    public static QuestionDto QuestionToDto(Question question) {
        QuestionDto q = new QuestionDto();

        q.question = question.getQuestion();
        q.correctOption = question.getCorrectOption();
        q.wrongOptions = question.getWrongOptions();
        q.difficulty = question.getDifficulty();

        return q;
    }

    public static Question QuestionDtoToQuestion(QuestionDto questionDto) {
        Question q = new Question();

        q.setQuestion(questionDto.question);
        q.setCorrectOption(questionDto.getCorrectOption());
        q.setWrongOptions(questionDto.wrongOptions);
        q.setDifficulty(questionDto.difficulty);

        return q;
    }

}

File: node-backend/src/Common/Mapper/QuestionMapper.ts
import { Question, QuestionWrongOption, QuestionDTO } from '../../Types/Question';

const mapQuestionToDto = (
  question: Question,
  wrongOptionsArray: QuestionWrongOption[]
): QuestionDTO => {
  const wrongOptions: [string, string, string] = [
    wrongOptionsArray[0].wrong_option,
    wrongOptionsArray[1].wrong_option,
    wrongOptionsArray[2].wrong_option
  ];

  const questionDTO: QuestionDTO = {
    correct_option: question.correct_option,
    difficulty: question.difficulty,
    id: question.id,
    question: question.question,
    wrong_options: wrongOptions
  };

  return questionDTO;
};

export default {
  mapQuestionToDto
};
